### DISCUSSION TITLE
Node-live-example doesn't work for safari.

### DISCUSSION THREAD

**Discussion by gziz** [deepgram/community#222] [General help]
I'm running the Node-live-example [https://github.com/deepgram-devs/node-live-example.git](https://github.com/deepgram-devs/node-live-example.git), which is a server-client in node.js to run streaming transcription with audio from the browser's mic.

I'm testing the code exactly as it is in the repo.
It runs as expected in Chrome and edge. 
The `transcriptReceived` listener receives the following JSON:
```
{
  channel_index: [ 0, 1 ],
  duration: 1.4199998,
  start: 1.95,
  is_final: true,
  speech_final: true,
  channel: { alternatives: [ [Object] ] },
  metadata: {
    request_id: '7c679594-e1f9-48c0-a66a-9e66b7be2855',
    model_info: { name: 'general-nova', version: '2023-03-13.31000', arch: 'nova' },
    model_uuid: '3312b52b-d08e-4072-bfff-124c410b770d'
  }
}
```

However, for safari, this is what I get:
```
{
  transaction_key: 'deprecated',
  request_id: 'ecc5c455-7d6f-4763-be48-80f7f19d2584',
  sha256: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855',
  created: '2023-06-19T18:57:15.719Z',
  duration: 0,
  channels: 0
}
```
Has anyone been through this before?
Thanks!

**Comment by lukeocodes** [comment]
You may want to check that MediaRecorder returns something. It may be that you need a polyfill in Safari. I will make a note to update that sample code.

**Reply by gziz** [reply]
Got it! I checked, it does return something, however instead of being 'audio/webm' as in Chrome, it's 'audio/mp4'. I'm assuming this is the problem. I will take a look at polyfill. Cause it seems this is a safari problem.

**Reply by gziz** [reply]
It cannot find a way to record a compatible streaming audio format in Safari (compatible with Deepgram streaming).
Maybe I must receive the audio in mp4 and convert it in server to a compatible coding. Any thoughts?

**Reply by gziz** [reply]
@lukeocodes Have there been any modifications? I'm now getting the same error `transaction_key: 'deprecated',` with the browsers that used to work. Thanks.

**Reply by raxitjat2343** [reply]
@lukeocodes i have the same error in all browsers,any solution?

**Reply by drdavidg** [reply]
Any update on this issue? Safari browser is returning `audio/mp4` streaming raw audio. I cannot find a solution to get that into an audio encoding codec supported by Deepgram.

**Comment by ophir11235813** [comment]
The fix was simple for me: Safari seems to send empty data to Deepgram servers while Chrome doesn't. So I added this to my client:
this.microphone.ondataavailable = (e) => {
      if (e.data.size > 0) {
        // Check if the data packet is not empty
        const data = e.data;
        console.log("client: sending data to websocket");
        this.socket.send(data);
      } else {
        console.warn("Empty data packet received, not sending to WebSocket.");
      }
    };
    
This fixed it on desktop, though I haven't tried it on mobile yet. Hope this helps you!
